### ISSUE TITLE
Create a linux AppImage client package

### ISSUE THREAD

**Issue by nixfu** [bluecherrydvr/bluecherry-client#328]
Since you guys seem great about trying to maintain Linux support in the client, I would suggest you look into creating a Linux AppImage package.   

It makes supporting multiple Linux distros MUCH easier.

https://appimage.org/

This actually might be a way to package the existing client to work on newer systems that come with QT5 now but still make them able to run the client because the AppImage can contain its own QT4 libs inside it.

**Comment by antonsviridenko** [comment]
Porting to Qt5 is already in progess and should be finished soon.

**Comment by nixfu** [comment]
This also eliminates the need to create debs, rpms, etc.   An AppImage can basically run on any linux system.

**Comment by antonsviridenko** [comment]
Yes, but I personally pefer more traditional approach to packaging :)

**Comment by nixfu** [comment]
Fair enough, but you don't support non ubtuntu/debian today and this would make you able to support almost any distro.  I have been using more and more 3rd party apps as AppImages and finding it a great solution for teams that have to support Linux clients.

I guess I will be waiting until its done porting to Qt5 because I have to compile it myself since I don't use a deb based Linux system.

**Comment by curtishall** [comment]
Why would AppImage preferred over something like snap - https://snapcraft.io/?

(I'm not trying to be difficult, I'm curious.)

**Comment by nixfu** [comment]
SNAP's (and Flatpack) are ok, but I find them a hassle because they require daemons and extra software to be installed to use them, and some other support like a ton of nested filesystems to be mounted etc.      I

I personally find AppImage to be better because it is 100% self-contained with zero dependency issues, and no requirements other than to download the .AppImage, set to executable and run it.  It is basically the Linux equivalent of what is a "portable app" on Windows.

AppImage is becoming the most common way for things like CryptoCoin wallet programs to be distributed because it is more trustworthy to get it directly from the developer than SNAP's etc that are distributed out of repos.

**Comment by tstivers1990** [comment]
> Why would AppImage preferred over something like snap - https://snapcraft.io/?
> 
> (I'm not trying to be difficult, I'm curious.)

Flatpak is fine. Snap, however, is vendor lock in from Canonical. There can only be one snap store, unlike Flatpak where the server code is available and anyone can create their own Flatpak store. If you're going to target one, I would target Flatpak, as you can install Flatpak on any distro you can install Snap on anyway.
